packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,734 @@
+    // or the field name has dashes in it and its schema key was converted to underscores during initiation.
+    if (!scheme) {
+      const underscoredModelKey = parent ? `${parent}-${key.replace(/-/g, '_')}` : key.replace(/-/g, '_')
+
+      // Here we check if the underscored key exists in the schema and the `fieldNameHasDashes` flag is true
+      if (props.schema[underscoredModelKey]?.fieldNameHasDashes) {
+        modelKey = underscoredModelKey
+        scheme = props.schema[modelKey]
+      }
+    }
+
+    const value = data[key]
+    const type = typeof value
+
+    // Ensure Object Advanced field is saved to model as empty object
+    if (scheme && scheme.type === 'object-advanced' && !value) {
+      formModel[modelKey] = {}
+      originalModel[modelKey] = {}
+
+      return
+    }
+
+    // avoid pass by ref
+    if (Array.isArray(value)) {
+      formModel[modelKey] = JSON.parse(JSON.stringify(value))
+      originalModel[modelKey] = JSON.parse(JSON.stringify(value))
+
+      return
+    }
+
+    // update the model for each child of the object
+    if (type === 'object' && value && !value.length) {
+      return updateModel(value, modelKey)
+    }
+
+    // treatment for arrays, convert to comma separated string
+    if (type === 'object' && value && value.length && scheme.type === 'input') {
+      formModel[modelKey] = value.join(', ')
+      originalModel[modelKey] = value.join(', ')
+
+      return
+    }
+
+    formModel[modelKey] = value
+    originalModel[modelKey] = value
+  })
+
+  emit('model-updated', {
+    model: formModel,
+    originalModel,
+    data: getModel(),
+  })
+}
+
+const loading = ref(true)
+const initFormModel = (): void => {
+  if (props.record && props.schema) {
+    // global fields
+    updateModel({
+      enabled: props.record.enabled ?? true,
+      ...(props.record.instance_name && { instance_name: props.record.instance_name || '' }),
+      ...(props.record.protocols && { protocols: props.record.protocols }),
+      ...(props.record.tags && { tags: props.record.tags }),
+    })
+
+    if (props.record.data) {
+      updateModel(props.record.data)
+    } else if (props.record.config) {
+      // scope and top level fields
+      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
+        updateModel({
+          service_id: props.record.service_id,
+          route_id: props.record.route_id,
+          consumer_id: props.record.consumer_id,
+          consumer_group_id: props.record.consumer_group_id,
+          enabled: props.record.enabled,
+        })
+      }
+
+      // main plugin configuration
+      updateModel(props.record.config, 'config')
+    }
+  }
+
+  // scoping logic, convert _ to - for form model
+  // Check if incoming field exists in current model and if so update
+  if (entityIdField.value && props.entityId && props.schema) {
```

Comment:
This `if` block is for creating a plugin from within a specific entity, right? If so, we need to change the condition:
```suggestion
  if (entityIdField.value && props.entityId && props.schema && !props.editing) {
```

Because in the above scenario (editing a plugin that is scoped to both a service and a route), `entityIdField.value` is `route_id`, and `props.entityId` is the ID of the service (due to the order of route and service when assigning `entity` and `id` [here](https://github.com/Kong/public-ui-components/blob/fe14ce40824b00b92ec37ca1bba2c47b55579e8b/packages/entities/entities-plugins/src/components/PluginForm.vue#L243-L261)), so this block will assign the service ID to `updateFields.route_id`, which confuses the scope input.